offload/include/Shared/Types.h

```diff
@@ -0,0 +1,22 @@
+//===-- Shared/Types.h - Type defs shared between host and device - C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Environments shared between host and device.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef OMPTARGET_SHARED_TYPES_H
+#define OMPTARGET_SHARED_TYPES_H
+
+#ifndef OMPTARGET_DEVICE_RUNTIME
+#include <cstdint>
+#else
+#include "DeviceTypes.h"
+#endif
```

Comment:
The fact that we're redefining the standard integers here is simply us building this incorrectly, you can include `stdint.h` just fine from the GPU if you pass `-nostdlibinc` or `-ffreestanding`, or manually undefine `__STDC_HOSTED__`, in order of desirability.

Reply:
This is beyond the patch but we can absolutely remove the redefines afterwards.